src/main/java/org/sonarlint/intellij/ui/SonarLintRulePanel.kt

```diff
@@ -126,6 +127,8 @@ class SonarLintRulePanel(private val project: Project, private val parent: Dispo
 
         setLoadingText("Loading rule description...")
         clear()
+
+        ApplicationManager.getApplication().messageBus.connect().subscribe(LafManagerListener.TOPIC, LafManagerListener { updateUiComponents() })
```

Comment:
The last small improvement that you can do here is to connect to the bus by passing the `parent` disposable existing in this class. The listener will have a shorter lifecycle

Reply:
Done, there ya go mate 😄